app/services/course_service.py

```diff
@@ -1,18 +1,65 @@
 from typing import List
+from pydantic import BaseModel
 from sqlalchemy.orm import Session
-from app.models import CourseModel, InstructorModel
+from sqlalchemy.orm.exc import MultipleResultsFound, NoResultFound
+from app.models import CourseModel
 from app.schemas import CourseWithInstructorsSchema
-from app.core.config import settings
-from app.services import InstructorService
+from app.core.exceptions import MultipleCoursesExistException, NoCourseExistsException, CourseAlreadyExistsException
 
 class CourseService:
     def __init__(self, session: Session):
         self.session = session
     
     async def get_course(self) -> CourseModel:
-        return self.session.query(CourseModel).first()
+        try:
+            return self.session.query(CourseModel).one()
+        except MultipleResultsFound as e:
+            raise MultipleCoursesExistException()
```

Comment:
If multiple courses get created, there is no possible way to fetch one course via the API with this. I'd be wary of that, because I think we also dont have a way for instructors to delete a course if they make one by mistake

Reply:
By design, there should always be exactly one course in the database. If there are 0, it will break other components because we consider this assumption to be true. We could go ahead and change this and add special handling everywhere that we've assumed this, but it's reasonable enough to just create a template of the course before standing up the API, and then the Professor can modify that course instead of deleting it and creating a new one.

I'm not totally opposed to allowing multiple courses, so the instructor can create a new one then delete the old one. Or just set one to "active". However, I don't see that use case being relevant at all.

Reply:
Also to be clear, professors don't have the permission to create/delete courses, so there's no risk involved there unless we give them that permission.